test/gtest/uct/test_peer_failure.cc

```diff
@@ -566,37 +567,60 @@ class test_uct_peer_failure_keepalive : public test_uct_peer_failure
         test_uct_peer_failure::kill_receiver();
     }
 
-protected:
-    ucs::ptr_vector<ucs::scoped_setenv> m_env;
-};
+    void test_ep_check(bool post_am)
+    {
+        ucs_status_t status;
 
-UCS_TEST_SKIP_COND_P(test_uct_peer_failure_keepalive, killed,
-                     !check_caps(UCT_IFACE_FLAG_EP_CHECK))
-{
-    ucs_status_t status;
+        scoped_log_handler slh(wrap_errors_logger);
+        flush();
+        EXPECT_EQ(0, m_err_count);
 
-    scoped_log_handler slh(wrap_errors_logger);
-    flush();
-    EXPECT_EQ(0, m_err_count);
+        for (unsigned i = 0; i < 8; ++i) {
+            status = uct_ep_check(ep0(), 0, NULL);
+            ASSERT_UCS_OK(status);
+            flush();
+            /* allow keepalive request to complete */
+            short_progress_loop();
+        }
 
-    status = uct_ep_check(ep0(), 0, NULL);
-    ASSERT_UCS_OK(status);
-    flush();
+        /* we are still alive */
+        EXPECT_EQ(0, m_err_count);
+
+        kill_receiver();
 
-    /* allow keepalive requests to complete */
-    short_progress_loop();
+        status = uct_ep_check(ep0(), 0, NULL);
+        ASSERT_UCS_OK(status);
 
-    /* we are still alive */
-    EXPECT_EQ(0, m_err_count);
+        if (post_am) {
+            const uint64_t send_data = 0;
+            status = uct_ep_am_short(ep0(), 0, 0, &send_data, sizeof(send_data));
+            if (m_err_count == 0) {
+                ASSERT_UCS_OK(status);
+            }
```

Comment:
looks weird
if we did kill_receiver why would m_err_count be 0 ?

Reply:
the transport could not be fast enough to detect an error, e.g. some of IB transports

Reply:
so maybe wait until it's detected? since this test is supposed to test KA

Reply:
so maybe wait before posting?
or i don't understand the point of posting here? maybe add a comment?